platform/ios/src/MGLMapView.mm

```diff
@@ -1223,6 +1258,12 @@ - (void)handlePanGesture:(UIPanGestureRecognizer *)pan
         {
             CGPoint offset = CGPointMake(velocity.x * self.decelerationRate / 4, velocity.y * self.decelerationRate / 4);
             _mbglMap->moveBy({ offset.x, offset.y }, MGLDurationInSeconds(self.decelerationRate));
+            __weak MGLMapView *weakSelf = self;
```

Comment:
Good point. In that case, perhaps `-notifyGestureDidEndWithDrift:` is what you want. If not, try factoring out the animation into a similar method.